jobsdb/unionQuery.go

```diff
@@ -83,17 +85,28 @@ func (mj *MultiTenantHandleT) GetAllJobs(workspaceCount map[string]int, params G
 	dsList := mj.getDSList(false)
 	outJobs := make([]*JobT, 0)
 
+	if params.PayloadSize != nil {
+		if *(params.PayloadSize) <= 0 {
+			return outJobs
+		}
+	}
+
+	workspacePayloadLimitMap := make(map[string]int64)
+	for workspace, count := range workspaceCount {
+		workspacePayloadLimitMap[workspace] = int64(float64(int64(count)*(*(params.PayloadSize))) / float64(params.JobCount))
```

Comment:
What is the logic behind the payload limit that we are trying to impose? What is the meaning of `params.PayloadSize`? The total maximum payload size of all messages or the maximum payload size of a single message? Why would we want to limit the maximum payload size of a single message?

Assuming that we have 10 workspaces and each workspace has 2 jobs (10%). If my total payload limit is 10MB, shouldn't I roughly set a maximum payload size of 1MB (10MB*10%) per workspace?

Reply:
>What is the meaning of params.PayloadSize?

it's a limit on the total payloads read from the db.

> shouldn't I roughly set a maximum payload size of 1MB (10MB*10%) per workspace?

That's the plan. each workspace gets a limit on payload size in proportion to the number of jobs we want to pick for that workspace.

Reply:
I must have been confused then by the too many casts, isn't there anything we could do to simplify? e.g.

```
percentage := float64(count)/float64(params.JobCount)
payloadLimit := percentage*float64(*params.PayloadSize)
workspacePayloadLimitMap[workspace] = int64(payloadLimit)
```